q2_feature_table/_summarize/_vega_spec.py

```diff
@@ -0,0 +1,326 @@
+# ----------------------------------------------------------------------------
+# Copyright (c) 2016-2019, QIIME 2 development team.
+#
+# Distributed under the terms of the Modified BSD License.
+#
+# The full license is in the file LICENSE, distributed with this software.
+# ----------------------------------------------------------------------------
+
+import json
+
+
+def vega_spec(sample_metadata, sample_frequencies):
+
+    values = []
+    if sample_metadata:
+        sample_metadata = sample_metadata.filter_ids(
+            sample_frequencies.index)
+        df = sample_metadata.to_dataframe()
+
+        for i, row in df.iterrows():
+            values.append({
+                'id': i,
+                'metadata': {j: row[j] for j in df.columns},
+                'frequency': sample_frequencies[i]
+            })
+
+        metadata_categories = list(df.columns.values)
+        max_frequency = int(max(sample_frequencies.values.tolist()))
+
+    else:
+        for freq in sample_frequencies:
+            # add ids
+            values.append({
+                'frequency': i
+            })
+        metadata_categories = []
+        max_frequency = 0
+
+    spec = {
+            "$schema": "https://vega.github.io/schema/vega/v4.json",
+            "autosize": {"contains": "content",
+                         "type": "fit-x",
+                         "resize": True},
+            "width": 800,
```

Comment:
Just leaving a note here, we need to fix the resize issue. I will take a closer look once I submit this review.